### PR TITLE
Add a clang-format pre-commit and workflow

### PR DESCRIPTION
**Pull request by ASKabalan**
Make sure that all files are formatted correctly using the .clang-format file.

### CODE REVIEW

File: .github/workflows/formatting.yml [hunk 1]
```diff
@@ -19,3 +19,19 @@ jobs:
         python -m pip install pre-commit
     - name: Run pre-commit
       run: python -m pre_commit run --all-files
+  formatting-check:
```
Comment (EiffL) [inline .github/workflows/formatting.yml:22]: do we need this? aren't these formatting checks already exectude with the pre-commit hook?

**Review by EiffL — COMMENTED** [review]
It looks good to me, only question, is it true that yapf is not going to conflict with the other formatter? I think last time I ran it yapf may have also changed some c files, if that's the case, we just want to limit yapf to .py files

**Comment by EiffL** [comment]
sorry, just to confirm, in the github action, isn't the formatting check already ran by the pre-commit hook? is your added action code doing additional things?

**Comment by ASKabalan** [comment]
No it does the same thing.
Isn't that what it does for python/yapf?

**Comment by EiffL** [comment]
Then... we don't need another section in actions, you can see in the tests that the pre-commit already runs a clang formatting check:
image
It's probably to do everything through precommit because it ensures that there is no mismatch between the tests being run online, and the ones run locally.

Can you remove the added section you have in the action?

**Comment by ASKabalan** [comment]
Done

**Review by EiffL — APPROVED** [review]
Awesome :-)

**Pull request merged**